lib/guardian/post_guardian.rb

```diff
@@ -157,6 +157,7 @@ def can_edit_post?(post)
     return false if Discourse.static_doc_topic_ids.include?(post.topic_id) && !is_admin?
 
     return true if is_admin?
+    return false unless can_see_post_topic?(post)
```

Comment:
```suggestion
    return false if !can_see_post_topic?(post)
```

:P I will never not do this

Reply:
```suggestion
    return false if cannot_see_post_topic?(post)
```

Just for fun but I do wonder if we can have some magic in guardian that supports a `cannot_see_xyz` if a `can_see_xyz` method has been defined.

Reply:
> Just for fun but I do wonder if we can have some magic in guardian that supports a `cannot_see_xyz` if a `can_see_xyz` method has been defined.

Yes. This would be easy to implement using `#method_missing` if it's something we want. 🙂